src/tracer.js

```diff
@@ -1,15 +1,33 @@
-'use strict';
-
 import Span from './span';
 import SpanContext from './span_context';
-import Constants from './constants';
 import Reference from './reference';
+import * as Constants from './constants';
+import * as noop from './noop';
 
 /**
  * Tracer is the entry-point between the instrumentation API and the tracing
  * implementation.
  *
  * The default object acts as a no-op implementation.
+ *
+ * ### Note to implementators:
+ *
+ * An implementation should derive from Tracer. The base class provides
+ * "public" methods such as `startSpan()` which do two things before calling
+ * the a similarly named worker method (in this case, `_startSpan()`), which the
+ * derived class is expected to implement. The base does two things before
+ * calling the worker method:
+ *
+ * (1) It normalizes the arguments before passing the arguments along to the
+ * derived class. This avoids all N implementations from having to re-implement
+ * error-prone argument manipulation to support different call signatures.  For
+ * example, `inject()` allows either a `Span` or a `SpanContext` to be passed to
+ * the function; `_inject()` will always be passed a `SpanContext`.
```

Comment:
I'm not sure I understand the benefit here. Why not just have a single versioned API spec to implement to?